sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -398,6 +398,15 @@ def test_one_hot_encoder_feature_names_drop(drop, expected_names):
     assert_array_equal(expected_names, feature_names)
 
 
+def test_one_hot_encoder_drop_equals_if_binary():
+    X = [['Male', 1], ['Female', 3], ['Female', 2]]
+    expected = np.array([[1., 1., 0., 0.], [0., 0., 0., 1.], [0., 0., 1., 0.]])
```

Comment:
lay this out over three lines so that it is easier to see what's going on.

Reply:
Made all the changes